Ability to match field names in a case sensitive manner

mflorin opened the pull request:
Fixes #208 
I've added a new configuration option and the associated code to be able to match map keys to field names case sensitively.
However, the first letter of struct field names is still compared case insensitively, to allow for the Go convention of publicly exported member names.

mitchellh commented:
Thanks! I think the "ignore first character" is a bit odd... though I understand why you did it. Can you add a test for this?

rpajak-solarwinds commented:
> Thanks! I think the "ignore first character" is a bit odd... though I understand why you did it. Can you add a test for this?

I agree with @mitchellh .

`|| strings.Title(mK) == fieldName` should be removed. 

@mitchellh any chances to have it soon? I want to use `CaseSensitive` together with `ErrorUnused` to have a very strict control during decoding.

mflorin commented:
Yes, agreed, the logic is odd :) I think @rpajak-solarwinds 's idea to have a config would work. I'll make the necessary changes.

mflorin commented:
I've added another config, `TitleCaseInsensitive` which works together with `CaseSensitive`. I've also added the tests. Let me know if that works better :) . Thanks.

mitchellh commented:
I went with #250 since that supports any custom name matching logic. Sorry, but thank you!
